subsys/net/ip/Kconfig.debug

```diff
@@ -91,7 +91,16 @@ config NET_DEBUG_NET_PKT
 	default n
 	select NET_BUF_POOL_USAGE
 	help
-	Enables printing of network packet and buffer allocations and frees.
+	Enables collection of network packet and buffer allocations and frees.
+
+config NET_DEBUG_NET_PKT_ALL
+	bool "Debug network packet and buffer individual allocation"
```

Comment:
My argument for the proposed patch is:

* existing NET_DEBUG_NET_PKT is kept but it is only collecting memory allocation information
* new NET_DEBUG_NET_PKT_ALL will print all the allocations, it depends on NET_DEBUG_NET_PKT

With these changes, we do not need to change much code. I tried the proposal that @tbursztyka had, but to my eyes this required more changes.